docs/rules/jsx-curly-brace-presence.md

```diff
@@ -0,0 +1,153 @@
+# Enforce curly braces or disallow unnecessary curly braces in JSX props and/or children. (react/jsx-curly-brace-presence)
+
+This rule allows you to enforce curly braces or disallow unnecessary curly braces in JSX props and/or children.
+
+For situations where JSX expressions are unnecessary, please refer to [the React doc](https://facebook.github.io/react/docs/jsx-in-depth.html) and [this page about JSX gotchas](https://github.com/facebook/react/blob/v15.4.0-rc.3/docs/docs/02.3-jsx-gotchas.md#html-entities).
+
+**Fixable:** This rule is automatically fixable using the `--fix` flag on the command line
+
+## Rule Details
+
+By default, this rule will check for and warn about unnecessary curly braces in both JSX props and children.
+
+You can pass in options to enforce the presence of curly braces on JSX props or children or both. The same options are available for not allowing unnecessary curly braces as well as ignoring the check.
+
+## Rule Options
+
+```js
+...
+"react/forbid-elements": [<enabled>, { "props": <string>, "children": <string> }]
+...
+```
+
+or alternatively
+
+```js
+...
+"react/forbid-elements": [<enabled>, <string>]
+...
+```
+
+### Valid options for <string>
+
+They are `always`, `always,single`, `always,double`, `always,orignal`, `never` and `ignore` for checking on JSX props and children.
+
+* `always`: always enforce curly braces inside JSX props or/and children and fix with double quotes inside the generated JSX expressions
+* `always,single`: always enforce curly braces and fix with single quotes
```

Comment:
so, first - i think rather than squashing the quote style into the string value, this should be an object.

However, quote style is something that another rule should handle. This one should just pick a default - i'd go with double quotes - and let the jsx-quotes rule handle correcting it.